extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskIOConfig.java

```diff
@@ -33,12 +35,23 @@
 
 public class KinesisIndexTaskIOConfig extends SeekableStreamIndexTaskIOConfig<String, String>
 {
-  public static final int DEFAULT_RECORDS_PER_FETCH = 4000;
   public static final int DEFAULT_FETCH_DELAY_MILLIS = 0;
 
+  /**
+   * Together with {@link KinesisIndexTaskTuningConfig#MAX_RECORD_BUFFER_MEMORY}, don't take up more than 200MB
+   * per task.
+   */
+  private static final int MAX_RECORD_FETCH_MEMORY = 100_000_000;
+
+  /**
+   * Together with {@link KinesisIndexTaskTuningConfig#RECORD_BUFFER_MEMORY_MAX_HEAP_FRACTION}, don't take up more
+   * than 15% of the heap.
+   */
+  private static final double RECORD_FETCH_MEMORY_MAX_HEAP_FRACTION = 0.05;
```

Comment:
Seems like a typo since 15% is mentioned above

Reply:
I believe the intention here is that `RECORD_BUFFER_MEMORY_MAX_HEAP_FRACTION + RECORD_FETCH_MEMORY_MAX_HEAP_FRACTION` makes up the 15% threshold.

Reply:
Sorry, I misread it.